File: MaximumSubarray.py
def maxSubArray(nums) -> int:
    maxSum = float('-inf')
    currentSum = 0

    for n in nums:
        currentSum += n
        maxSum = max(currentSum, maxSum)
        if currentSum < 0:
            currentSum = 0
    return maxSum

print(maxSubArray([-2,1,-3,4,-1,2,1,-5,4]))


    

File: CSharp/MaximumSubarray.cs
using System;

public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSum = Int32.MinValue;
        int currentSum = 0;

        foreach(var n in nums)
        {
            currentSum += n;
            if (currentSum > maxSum) {
                maxSum = currentSum;
            } 
            if (currentSum < 0) {
                currentSum = 0;
            } 
        }
        return maxSum;
    }
}

var nums = new int[] { -2, 1, -3, 4, -1, 2, 1, -5, 4 };
var solution = new Solution();
Console.WriteLine(solution.MaxSubArray(nums))


